crates/installer/src/engine.rs

```diff
@@ -85,37 +89,43 @@ impl Installer {
         let mut boot_parts = vec![];
         let mut system_parts = vec![];
         for disk in disks.iter().filter(|d| !parents.iter().any(|r| *r == d.path)) {
-            if let Ok(parts) = disk.partitions().await {
-                // Exclude partitions related to `/` partition
-                let parts = parts
-                    .into_iter()
-                    .filter(|p| !root_nodes.iter().any(|r| *r == p.path))
-                    .collect::<Vec<_>>();
-                if let Some(esp) = parts
-                    .iter()
-                    .find(|p| matches!(p.kind, disk::PartitionKind::ESP))
-                    .cloned()
-                {
-                    let xbootldr = parts
-                        .iter()
-                        .find(|p| matches!(p.kind, disk::PartitionKind::XBOOTLDR))
-                        .cloned();
-                    boot_parts.push(BootPartition {
-                        esp,
-                        xbootldr,
-                        parent_desc: disk.to_string(),
-                    })
+            let parts = match disk.partitions().await {
```

Comment:
Note: This bit of the diff becomes way more readable when ignoring whitespace changes.